lib/Dialect/SystemC/SystemCTypes.cpp

```diff
@@ -12,6 +12,8 @@
 
 #include "circt/Dialect/SystemC/SystemCTypes.h"
 #include "circt/Dialect/SystemC/SystemCDialect.h"
+#include "mlir/IR/DialectImplementation.h"
+#include "llvm/ADT/TypeSwitch.h"
```

Comment:
These seem unnecessary?

Reply:
`DialectImplementation.h` is necessary because the dialect's `parseType` and `printType` are also generated in `SystemCTypes.cpp.inc` and take a `DialectAsmParser` and `DialectAsmPrinter` as argument which are defined in that header file.
`TypeSwitch.h` is necessary because in `generatedTypeParser` and `generatedTypePrinter` a TypeSwitch is used to select the correct parser/printer for the given mnemonic.